### PR TITLE
feat: add proxy support for outputs/cloudwatch (#11388)

### LINKED ISSUE #11388: [outputs.cloudwatch] proxy support
## Feature Request
Add proxy config option for cloudwatch output plugin (input plugin received this a while ago: [#8904](https://github.com/influxdata/telegraf/issues/8904)).

### Proposal:
Additional option "http_proxy" for outputs.cloudwatch similiar to inputs.cloudwatch. It should also allow user credentials or respect environment vars.

### Current behavior:
Communication in proxy situation fails

### Desired behavior:
Communication is done used proxy

### Use case: 
We are trying to connect to aws cloudwatch from a containerized telegraf inside a company network through a proxy. Despite our best efforts cant get it workingr (efforts: /etc/environment into container, setting env vars for the container when starting). Everything works when no proxy is present.

### PR DESCRIPTION
**Pull request by scholz**
### Required for all PRs:

- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #11388

Used proxy code from inputs/cloudwatch to add proxy to outputs cloudwatch.
Did extensive (manual) testing to confirm functionality; will add details to issue.

### CODE REVIEW

**Comment by srebhan** [comment]
@scholz thanks for looking into this! Would it make sense to completely switch to [`common/HTTPClientConfig`](https://github.com/influxdata/telegraf/blob/3008d918b71c7826cee6e6a5855c28ed9ba23cd5/plugins/common/http/config.go#L20) to create the client? This way you could save even more code...

**Comment by scholz** [comment]
Ok I removed some code and switched to httpconfig. As said in the issue I am new to go so bear with me for any oddities I may have inserted.

Also note that it seems that now timeouts only happen in the httpclient create context and therefore errors looks different (and I think less informative) than before:

previously when the proxy host wasnt responding the error was this:
2022-06-28T13:00:12.175280261Z 2022-06-28T13:00:12Z E! [agent] Error writing to outputs.cloudwatch: operation error CloudWatch: PutMetricData, exceeded maximum number of attempts, 3, https response error StatusCode: 0, RequestID: , request send failed, Post "https://monitoring.eu-central-1.amazonaws.com/": proxyconnect tcp: dial tcp 212.62.95.45:1080: i/o timeout

now it's this:
2022-06-30T20:46:24.280273225Z 2022-06-30T20:46:24Z E! [agent] Error writing to outputs.cloudwatch: operation error CloudWatch: PutMetricData, exceeded maximum number of attempts, 3, https response error StatusCode: 0, RequestID: , request send failed, Post "https://monitoring.eu-central-1.amazonaws.com/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

What would be a good way to improve this? Would it help to reintroduce the timeout variable in the aws struct and define it to some shorter time than the ctx timeout?

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for the nice work @scholz!

Regarding the error message, I don't immediately see where the difference comes from, but it might make sense to track it down and fix it for `common/http`... Any ideas? If so, it should be a new PR... Thanks again for your contribution!

File: plugins/outputs/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -159,12 +164,26 @@ func (*CloudWatch) SampleConfig() string {
...
+	client, err := c.HTTPClientConfig.CreateClient(ctx, c.Log)
+
 	if err != nil {
 		return err
 	}
 
-	c.svc = cloudwatch.NewFromConfig(cfg)
+	c.svc = cloudwatch.NewFromConfig(cfg, func(options *cloudwatch.Options) {
+		// Disable logging
+		options.ClientLogMode = 0
```
Comment (reimda) [inline plugins/outputs/cloudwatch/cloudwatch.go:183]: It looks like disabling logging is unrelated to this PR's goal of adding proxy http connections. Is this necessary or could we just accept NewFromConfig's default?

**Review by powersj — APPROVED** [review]

**Pull request merged**